### ISSUE TITLE
Feature Request / Idea: ISO Support

### ISSUE THREAD

**Issue by Ronald-Diemicke** [JMBeresford/retrom#211]
Any plans to allow scanning for various CD image types? Wondering if this might be outside the scope of this project...

**Comment by JMBeresford** [comment]
@Ronald-Diemicke Can you elaborate? Do you mean box/disk artworks for a given game? And by scanning, do you mean scanning the disk for existing files present in the library or scanning metadata providers such as Retrom currently does for cover+background images?

**Comment by JMBeresford** [comment]
Looking at the edited title it seems you mean the disk image dumps, rather than any artworks. Currently Retrom already tracks all files in a games given directory/path -- emulator profiles in Retrom can then provide specific supported extensions they support, optionally. Perhaps I am misunderstanding what you would like to see?

**Comment by Ronald-Diemicke** [comment]
Absolutely... So, I need to sit down actually TRY this all out, but my understanding is that if you setup the server end, it scans for all the various types of installable formats, then the client downloads it and runs it... which is great... 

However, if the installable media is a disc image, then it'd be neat to have the client download it and _mount it_ as a disc image and then install it...

**Comment by JMBeresford** [comment]
In general, when "installing" dumps in the Retrom client it simply downloads a local copy that the client manages. Then you would have an emulator configured to launch said dump. In what circumstances would mounting the image be required? Are you referring to native ( i.e. PC, macOS or Linux ) games? Or perhaps emulators that require a mounted/inserted disc to launch from?

**Comment by Ronald-Diemicke** [comment]
More referring to native ( i.e. PC, macOS or Linux ) games. 

I'm using this less for emulation right now (using a similar project for that https://github.com/rommapp/romm) - but if this does what I'm looking for, it could solve a problem I have where I'd like to make my PC/Linux native game library installable on my steamdeck easily...

**Comment by JMBeresford** [comment]
Understood, this is something I plan to support soon. I've been thinking on how best to make the process ergonomic and intuitive, as it is not an easily problem with a one-size-fits-all solution. Mounting installation ISOs rather then using the installers from a dump/repack is yet another edge-case I'll need to consider, as I have not considered this previously.

Rest assured that this is relatively high on my list of priorities, though!

**Comment by Ronald-Diemicke** [comment]
Cool. No worries. I suspect this will be a PITA to do cross-platform given all the of the various disc image types... but glad it's on the radar!
